convert_hf_to_gguf.py

```diff
@@ -2702,6 +2702,80 @@ class StarCoder2Model(Model):
     model_arch = gguf.MODEL_ARCH.STARCODER2
 
 
+@Model.register("Rwkv6ForCausalLM")
+class RwkvModel(Model):
+    model_arch = gguf.MODEL_ARCH.RWKV
+
+    def set_vocab(self):
+        assert (self.dir_model / "rwkv_vocab_v20230424.txt").is_file()
+        vocab_size = self.hparams.get("vocab_size", 65536)
+
+        tokens: list[bytes] = ['<s>'.encode("utf-8")]
+        toktypes: list[int] = [gguf.TokenType.CONTROL]
+
+        with open(self.dir_model / "rwkv_vocab_v20230424.txt", "r", encoding="utf-8") as f:
+            lines = f.readlines()
+            for line in lines:
+                x = eval(line[line.index(' '):line.rindex(' ')])
```

Comment:
This can execute abritrary code, as long as it's in a file named `rwkv_vocab_v20230424.txt`.

Use [`ast.literal_eval`](https://docs.python.org/3/library/ast.html#ast.literal_eval) instead (by also adding `import ast` in this file) to avoid allowing to execute arbitrary expressions, while still parsing strings.

I think `line.split(' ')` could likely be used instead of using `line.index(' ')` and `line.rindex(' ')`.

Something like

```python3
parts = line.split(' ')
assert len(parts) >= 3
token_id, token, token_len = int(parts[0]), ast.literal_eval(' '.join(parts[1:-1])), int(parts[-1])